File: jimmer-ddd-java-sample/domain/src/main/java/org/morecup/jimmerddd/java/sample/domain/base/SnowflakeIdGenerator.java
package org.morecup.jimmerddd.java.sample.domain.base;

import org.babyfish.jimmer.sql.meta.UserIdGenerator;

import java.lang.Class;

import static org.morecup.jimmerddd.java.sample.domain.base.SpringContextUtils.snowflakeManager;

public class SnowflakeIdGenerator implements UserIdGenerator<Long> {
    @Override
    public Long generate(Class<?> entityType) {
        return snowflakeManager().nextId();
    }
}

File: jimmer-ddd-kotlin-sample/domain/src/main/kotlin/org/morecup/jimmerddd/kotlin/sample/domain/base/SnowflakeIdGenerator.kt
package org.morecup.jimmerddd.kotlin.sample.domain.base

import org.babyfish.jimmer.sql.meta.UserIdGenerator
import org.morecup.jimmerddd.kotlin.sample.domain.base.DomainRegistry.snowflakeManager

class SnowflakeIdGenerator : UserIdGenerator<Long> {
    override fun generate(entityType: Class<*>): Long? {
        return snowflakeManager().nextId()
    }
}